docs/current_docs/guides/snippets/build-test-publish-java-spring/go/main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"context"
+)
+
+type MyModule struct{}
+
+func (m *MyModule) Build(ctx context.Context, source *Directory) (string, error) {
+
+	mariadb := dag.Mariadb().Serve(MariadbServeOpts{Version: "10.11.2", DbName: "petclinic"})
+
+	// TODO: this doesn't work because the docker CLI isn't installed. And
+	// installing it just gives us a broken podman CLI. Whee.
+	//
+	// Can we just disable the tests that use Docker?
```

Comment:
Interesting idea. I checked the app README, it says `At development time we recommend you use the test applications set up as main() methods in PetClinicIntegrationTests (using the default H2 database and also adding Spring Boot Devtools), MySqlTestApplication and PostgresIntegrationTests. These are set up so that you can run the apps in your IDE to get fast feedback and also run the same classes as integration tests against the respective database. The MySql integration tests use Testcontainers to start the database in a Docker container, and the Postgres tests use Docker Compose to do the same thing.`

https://github.com/spring-projects/spring-petclinic?tab=readme-ov-file#test-applications

Reply:
I got an example working using test containers. You shouldn't start a Maria db as a service but instead start docker as a service. You shouldn't need a docker client for test containers the Java code interacts with docker. I did have to disable ryuk. TESTCONTAINERS_RYUK_DISABLED=true